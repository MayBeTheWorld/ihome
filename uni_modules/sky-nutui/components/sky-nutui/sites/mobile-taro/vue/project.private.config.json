{
  "setting": {
    "compileHotReLoad": true
  },
  "description": "项目私有配置文件。此文件中的内容将覆盖 project.config.json 中的相同字段。项目的改动优先同步到此文件中。详见文档：https://developers.weixin.qq.com/miniprogram/dev/devtools/projectconfig.html",
  "condition": {
    "miniprogram": {
      "list": [
        {
          "name": "feedback/pages/notify/index",
          "pathName": "feedback/pages/notify/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "exhibition/pages/countdown/index",
          "pathName": "exhibition/pages/countdown/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "business/pages/signature/index",
          "pathName": "business/pages/signature/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "exhibition/pages/collapse/index",
          "pathName": "exhibition/pages/collapse/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "dentry/pages/textarea/index",
          "pathName": "dentry/pages/textarea/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        }
      ]
    }
  },
  "projectname": "NutUI-Taro"
}