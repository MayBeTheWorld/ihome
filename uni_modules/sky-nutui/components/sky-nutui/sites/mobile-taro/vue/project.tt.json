{
  "miniprogramRoot": "./",
  "projectname": "@nutui/nutui-taro-mobile",
  "appid": "touristappid",
  "setting": {
    "es6": false,
    "minified": false
  }
}
