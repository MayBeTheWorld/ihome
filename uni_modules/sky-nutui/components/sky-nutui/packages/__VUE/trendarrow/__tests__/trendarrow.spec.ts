import { config, DOMWrapper, mount } from '@vue/test-utils';
import Trendarrow from '../index.vue';
import NutIcon from '../../icon/index.vue';
import { nextTick, toRefs, reactive } from 'vue';

beforeAll(() => {
  config.global.components = {
    NutIcon
  };
});

afterAll(() => {
  config.global.components = {};
});

test('should render rate when used ', async () => {
  const wrapper = mount(Trendarrow, {
    props: {
      rate: 12.325
    }
  });
  expect(wrapper.find('.nut-trendarrow-rate').text()).toBe('12.33%');
});

test('should render digits when used ', async () => {
  const wrapper = mount(Trendarrow, {
    props: {
      rate: 12.325,
      digits: 1
    }
  });
  expect(wrapper.find('.nut-trendarrow-rate').text()).toBe('12.3%');
});

test('should render digits when used ', async () => {
  const wrapper = mount(Trendarrow, {
    props: {
      rate: 12.325,
      digits: 1
    }
  });
  expect(wrapper.find('.nut-trendarrow-rate').text()).toBe('12.3%');
});

test('should render showSign when used ', async () => {
  const wrapper = mount(Trendarrow, {
    props: {
      rate: 12.325,
      showSign: true
    }
  });
  expect(wrapper.find('.nut-trendarrow-rate').text()).toBe('+12.33%');
});
test('should render showZero when used ', async () => {
  const wrapper = mount(Trendarrow, {
    props: {
      rate: 0,
      showSign: true,
      showZero: true
    }
  });
  expect(wrapper.find('.nut-trendarrow-rate').text()).toBe('0.00%');
});
test('should not render 0  when showZero is false ', async () => {
  const wrapper = mount(Trendarrow, {
    props: {
      rate: 0,
      showSign: true,
      showZero: false
    }
  });
  expect(wrapper.find('.nut-trendarrow-rate').text()).toBe('--');
});
test('should render left icon when arrowLeft', async () => {
  const wrapper = mount(Trendarrow, {
    props: {
      rate: 10,
      arrowLeft: true
    }
  });
  expect(wrapper.findAll('.nut-trendarrow-icon-before').length).toBe(0);
  expect(wrapper.findAll('.nut-trendarrow-icon-after').length).toBe(1);
});
test('should render sync text color when syncTextColor is true', async () => {
  const wrapper = mount(Trendarrow, {
    props: {
      rate: 10,
      syncTextColor: true
    }
  });
  const span = wrapper.find<HTMLElement>('.nut-trendarrow-rate');
  expect(span.element.style.color).toContain('250, 44, 25');
});

test('should render text color when textColor used', async () => {
  const wrapper = mount(Trendarrow, {
    props: {
      rate: 10,
      textColor: 'rgb(39,197,48)',
      syncTextColor: false
    }
  });
  const span = wrapper.find<HTMLElement>('.nut-trendarrow-rate');
  expect(span.element.style.color).toContain('(39, 197, 48)');
});

test('should render triangle color when riseColor used', async () => {
  const wrapper = mount(Trendarrow, {
    props: {
      rate: 10,
      riseColor: 'rgb(73,143,242)'
    }
  });
  const span = wrapper.find<HTMLElement>('.nut-trendarrow-rate');
  expect(span.element.style.color).toContain('(73, 143, 242)');
});

test('should render triangle color when dropColor used', async () => {
  const wrapper = mount(Trendarrow, {
    props: {
      rate: -10,
      dropColor: 'rgb(73,143,242)',
      showSign: true
    }
  });
  const span = wrapper.find<HTMLElement>('.nut-trendarrow-rate');
  expect(span.element.style.color).toContain('(73, 143, 242)');
});
test('should render triangle size when set iconsize', async () => {
  const wrapper = mount(Trendarrow, {
    props: {
      rate: -10,
      iconSize: '14px'
    }
  });
  await nextTick();
  await nextTick();
  expect(wrapper.html()).toMatchSnapshot();

  const span = wrapper.find<HTMLElement>('.nut-icon');
  expect(span.element.style.fontSize).toContain('14px');
});
