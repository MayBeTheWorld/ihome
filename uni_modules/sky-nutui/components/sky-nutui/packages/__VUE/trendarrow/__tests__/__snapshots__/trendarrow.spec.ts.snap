// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render triangle size when set iconsize 1`] = `
"<view class=\\"nut-trendarrow\\"><span class=\\"nut-trendarrow-icon-before nut-trendarrow-rate\\" style=\\"color: rgb(100, 181, 120);\\">10.00%</span><i class=\\"nutui-iconfont nut-icon nut-icon-triangle-down\\" style=\\"color: rgb(100, 181, 120); font-size: 14px; width: 14px; height: 14px;\\" src=\\"\\"></i>
  <!--v-if-->
</view>"
`;
